pandas/core/indexes/base.py

```diff
@@ -4775,11 +4774,21 @@ def get_indexer_for(self, target, **kwargs):
         numpy.ndarray
             List of indices.
         """
-        if self.is_unique:
+        if self._index_as_unique:
             return self.get_indexer(target, **kwargs)
         indexer, _ = self.get_indexer_non_unique(target, **kwargs)
         return indexer
 
+    @property
```

Comment:
why can't we just use a cached property for .is_unique?

Reply:
either way.  is_unique is cached, so this lookup isnt expensive

Reply:
kk